api/types.go

```diff
@@ -141,27 +142,55 @@ type SeriesType string
 // end =   0 mod resolution
 //
 // This range is inclusive of Start and End (i.e. [Start, End]). Start and End
-// are Unix second timestamps. Resolution is in seconds.
-// TODO: Make these ms
+// are Unix milliseconds timestamps. Resolution is in milliseconds.
+// (Millisecond precision allows an effective range of 290 million years in each direction)
 type Timerange struct {
-	Start      int64
-	End        int64
-	Resolution int64
+	start      int64
+	end        int64
+	resolution int64
+}
+
+// DefaultTimerange is a valid timerange which can be safely used
+func DefaultTimerange() Timerange {
+	return Timerange{start: 0, end: 0, resolution: 30}
+}
+
+// Start() returns the .start field
+func (tr Timerange) Start() int64 {
+	return tr.start
+}
+
+// End() returns the .end field
+func (tr Timerange) End() int64 {
+	return tr.start
+}
+
+// Resolution() returns the .resolution field
+func (tr Timerange) Resolution() int64 {
+	return tr.start
+}
+
+func NewTimerange(start, end, resolution int64) (Timerange, error) {
+	result := Timerange{start: start, end: end, resolution: resolution}
+	if !result.isValid() {
+		return Timerange{}, errors.New("invalid timerange")
+	}
+	return result, nil
 }
 
 // IsValid determines whether the given timerange meets the constraint.
-func (tr Timerange) IsValid() bool {
-	return tr.Resolution > 0 &&
-		tr.Start%tr.Resolution == 0 &&
-		tr.End%tr.Resolution == 0 &&
-		tr.Start <= tr.End
+func (tr Timerange) isValid() bool {
```

Comment:
Should inline this in the constructor so we can have better error messaging for all the things that can go wrong.